### PR TITLE
Added Documentation for Defined Entity Types

### PR DESCRIPTION
**Pull request by AlexPasheva**

### CODE REVIEW

File: documentation/extensibility-platform/defined-entities/defined-entity-types.md [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+
+### NSS
+User defined field that is part of the identification.
+
+### Version
+Once an instance of a defined entity type has been created, the type, schema, and behaviors cannot be changed anymore. This is essential because if modifications are made, entities with the prior schema would become invalid. This is why types must be versioned and versions must follow [Smantic Versioning](https://semver.org/) principles.
+
+When a defined entity instance is based on an earlier version of the entity type, you can upgrade the defined entity to use a later version of the type by setting the type property of the entity to the ID of the new type. More information on versioning can be found here: [Versioning](//TODO:link).
+
+### Exact and Classification IDd
```
Comment (iiliev2) [inline documentation/extensibility-platform/defined-entities/defined-entity-types.md:43]: > Exact and Classification ID~~d~~

File: documentation/extensibility-platform/defined-entities/defined-entity-types.md [hunk 2]
```diff
@@ -0,0 +1,163 @@
+# Defined Entity Types
+
+Defined entity types describes the structure and behaviors of defined entities.
+
+The behaviors of a type are specified by associating one or more interfaces with it. At least one interface must be connected with a type.
```
Comment (iiliev2) [inline documentation/extensibility-platform/defined-entities/defined-entity-types.md:5]: > The behaviors of a type are specified by associating one or more interfaces with it.

This is rather vague. Types inherit the behaviors of their interface(s). Types may override one or more of those behaviors. Add a section explaining how to do that.

File: documentation/extensibility-platform/defined-entities/defined-entity-types.md [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+    },
+    "vendor": "vmware",
+    "interfaces": [],
+    "hooks": null,
+    "readonly": false
+}
+```
+#### Important Note: The vendor, nss and schema cannot be changed when updating the type.
+
+### Delete an interface
```
Comment (iiliev2) [inline documentation/extensibility-platform/defined-entities/defined-entity-types.md:158]: > Delete a~~n interface~~type

File: documentation/extensibility-platform/defined-entities/defined-entity-types.md [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+```
+DELETE /cloudapi/1.0.0/entityTypes/<defined-entity-type-id>
+```
+
+#### Important Note: Defined entity type cannot be deleted if it is marked as readOnly.
```
Comment (iiliev2) [inline documentation/extensibility-platform/defined-entities/defined-entity-types.md:163]: Mention that this is a system field, cannot be set by the user. It is similar to the field of the same name in the interfaces.

**Review by iiliev2 — APPROVED** [review]

**Review by theotonchev — APPROVED** [review]
The comments will be addressed in a separate pull request

**Pull request merged**